src/loader/loader.c

```diff
@@ -505,8 +543,21 @@ loader_process_utility_hook(PlannedStmt *pstmt, const char *query_string,
 		{
 			SecLabelStmt *stmt = castNode(SecLabelStmt, pstmt->utilityStmt);
 
+			/*
+			 * Since this statement can be in a dump output, we only print an
+			 * error on anything that doesn't looks like a sane distributed
+			 * UUID.
+			 */
 			if (stmt->provider && strcmp(stmt->provider, SECLABEL_DIST_PROVIDER) == 0)
-				ereport(ERROR, (errmsg("TimescaleDB label is for internal use only")));
+			{
+				const char *uuid = strchr(stmt->label, SECLABEL_DIST_TAG_SEPARATOR);
+				if (!uuid || strncmp(stmt->label, SECLABEL_DIST_TAG, uuid - stmt->label) != 0 ||
+					!is_uuid(&uuid[1]))
+					ereport(ERROR,
+							(errmsg("TimescaleDB label is for internal use only"),
+							 errdetail("Security label is \"%s\".", stmt->label),
+							 errhint("Security label has to be of format \"dist_uuid:<UUID>\".")));
```

Comment:
I don't think this hint is appropriate since dist_uuid is not something a user should set. If this error happens, then either the dist_uuid is corrupted or the code mistook another UUID for the dist_uuid. In neither case should the user try to change the format to fix this issue or add this label him/herself.

I would probably just get rid of the hint.

Reply:
Understanding why something is a problem and tracking down the issue is just as important as being able to fix it. This hint is intended for users to understand what is wrong, and not so much for being able to repair it, so I think there is value in having this hint for debugging purposes.